src/fileSink.ts

```diff
@@ -0,0 +1,124 @@
+import * as fsp from 'fs/promises';
+
+import { LogEvent, LogEventLevel } from './logEvent';
+import { Sink } from './sink';
+
+export enum FileSize {
+  ONE_KB = 1024,
+  ONE_MB = FileSize.ONE_KB * 1024,
+  ONE_GB = FileSize.ONE_MB * 1024,
+}
+
+export interface FileSinkOptions {
+  fileName?: string;
+  outputDir?: string;
+  logEventLevel?: LogEventLevel;
+  maxFileSize?: number;
+}
+
+export const defaultFileSinkOptions: FileSinkOptions = {
+  outputDir: './logs',
+  logEventLevel: LogEventLevel.verbose,
+  maxFileSize: FileSize.ONE_MB * 10,
+};
+
+export class FileSink implements Sink {
+  protected options: FileSinkOptions;
+  protected name: string;
+  private content: string[];
+  private fileSystem: typeof fsp;
+
+  constructor(options: FileSinkOptions, fileSystem = fsp) {
+    this.options = {
+      ...defaultFileSinkOptions,
+      ...(options || {})
+    };
+    this.name = this.options.fileName || new Date().toISOString().split('T')[0];
+    this.content = [];
+    this.fileSystem = fileSystem;
+  }
+
+  public async emit(events: LogEvent[]) {
+    if (this.options.logEventLevel === LogEventLevel.off) return;
+
+    await this.manageFiles();
+
+    for (const event of events) {
+      if (this.options.logEventLevel === LogEventLevel.verbose || this.options.logEventLevel === event.level) {
```

Comment:
@davisb10 shouldn't this be changed (along with the default value of `logEventLevel`) so that it works the same way the `restrictedToMinimumLevel` of the ConsoleSink does?